adam/curriculum/phase2_curriculum.py

```diff
@@ -26,70 +67,164 @@
-            ],
-            syntax_hints=[IGNORE_COLORS],
+
+    templates = []
+    for cup in [CUP, CUP_2, CUP_3, CUP_4]:
+        cup_obj = standard_object("cup", cup)
+        liquid_0 = object_variable("liquid_0", required_properties=[LIQUID])
+        person_0 = standard_object("person_0", PERSON)
+
+        templates.append(
+            Phase1SituationTemplate(
+                "drink-cup",
+                salient_object_variables=[liquid_0, person_0],
+                actions=[
+                    Action(
+                        DRINK,
+                        argument_roles_to_fillers=[(AGENT, person_0), (THEME, liquid_0)],
+                        auxiliary_variable_bindings=[(DRINK_CONTAINER_AUX, cup_obj)],
+                    )
+                ],
+            )
         )
-        for cup in [CUP, CUP_2, CUP_3, CUP_4]
-    ]
 
     return phase2_instances(
-        "each cup by itself",
+        "drink - cup",
         chain(
             *[
                 all_possible(
                     cup_template,
                     chooser=PHASE1_CHOOSER_FACTORY(),
                     ontology=GAILA_PHASE_2_ONTOLOGY,
                 )
-                for cup_template in cup_templates
+                for cup_template in templates
             ]
         ),
         perception_generator=perception_generator,
     )
+
+
+def _make_put_in_curriculum() -> Phase1InstanceGroup:
+    agent = standard_object("agent", THING, required_properties=[ANIMATE])
+    theme = standard_object("theme", INANIMATE_OBJECT)
+    goal_in = standard_object("goal_in", INANIMATE_OBJECT, required_properties=[HOLLOW])
+
+    return phase1_instances(
+        "Capabilities - Put in",
+        sampled(
+            _put_in_template(agent, theme, goal_in, []),
+            ontology=GAILA_PHASE_1_ONTOLOGY,
+            chooser=PHASE1_CHOOSER_FACTORY(),
+            max_to_sample=20,
+        ),
+    )
+
+
+def build_gaila_m8_curriculum() -> Sequence[Phase1InstanceGroup]:
+    return list(
+        chain(
+            [
+                _make_plural_objects_curriculum(),  # plurals
+                _make_sit_on_chair_curriculum(),  # functionally defined objects
+                _make_drink_cups_curriculum(),
+                _make_pass_curriculum(),  # Subtle verb distinctions
+                _make_generic_statements_curriculum(),  # Generics
+                _make_part_whole_curriculum(),  # Part whole
+                _make_transitive_roll_curriculum(),  # External Limitations
+                _make_put_in_curriculum(),
+            ],
+            list(make_imprecise_temporal_descriptions()),  # Imprecise descriptions
+            list(make_subtle_verb_distinctions_curriculum()),  # Subtle verb distinctions
+            make_verb_with_dynamic_prepositions_curriculum(),  # Dynamic prepositions
+            make_prepositions_curriculum(),  # Relative prepositions
+            make_imprecise_size_curriculum(),
```

Comment:
If we add this to the end we don't mess the numbering of the output HTML files up for the corresponding document
```suggestion
            make_verb_with_dynamic_prepositions_curriculum(),  # Dynamic prepositions
            make_prepositions_curriculum(),  # Relative prepositions
            make_imprecise_size_curriculum(),
            list(make_subtle_verb_distinctions_curriculum()),  # Subtle verb distinctions
```